geopandas/sindex.py

```diff
@@ -764,19 +750,40 @@ def query_bulk(self, geometry, predicate=None, sort=False):
             return res
 
         @doc(BaseSpatialIndex.nearest)
-        def nearest(self, geometry):
+        def nearest(
+            self, geometry, return_all=True, max_distance=None, return_distance=False
+        ):
             if not compat.PYGEOS_GE_010:
                 raise NotImplementedError("sindex.nearest requires pygeos >= 0.10")
 
             geometry = self._as_geometry_array(geometry)
-            return super().nearest(geometry)
 
-        @doc(BaseSpatialIndex.nearest_all)
-        def nearest_all(self, geometry, max_distance=None, return_distance=False):
-            geometry = self._as_geometry_array(geometry)
-            return super().nearest_all(
+            if not return_all and max_distance is None and not return_distance:
+                return super().nearest(geometry)
+
+            result = super().nearest_all(
                 geometry, max_distance=max_distance, return_distance=return_distance
             )
+            if return_distance:
+                indices, distances = result
+            else:
+                indices = result
+
+            if not return_all:
+                # first subarray of geometry indices is sorted, so we can use this
+                # trick to get
+                uniques = np.diff(indices[0, :])
+                # always select the first element
+                uniques = np.insert(uniques, 0, 1)
```

Comment:
```suggestion
                uniques = np.insert(uniques, 0, 1).astype('bool')
```

I had to convert uniques to bool to get boolean indexing working properly, and this fixed the failing tests for me locally.

Reply:
Ah, yes, it was seeing it as indices .. Thanks!